zeroconf/__init__.py

```diff
@@ -2197,18 +2197,24 @@ def remove_all_service_listeners(self) -> None:
             self.remove_service_listener(listener)
 
     def register_service(
-        self, info: ServiceInfo, ttl: Optional[int] = None, allow_name_change: bool = False
+        self,
+        info: ServiceInfo,
+        ttl: Optional[int] = None,
+        allow_name_change: bool = False,
+        cooperating_responders: bool = False,
     ) -> None:
         """Registers service information to the network with a default TTL.
         Zeroconf will then respond to requests for information for that
         service.  The name of the service may be changed if needed to make
-        it unique on the network."""
+        it unique on the network. Additionally multiple cooperating responders
+        can register the same service on the network for resilience
+        (if you want this behavior set `cooperating_responders` to `True`)."""
```

Comment:
Should we state that we're not exactly RFC compliant without it (and, well, with it probably as well)?

Reply:
Hmm, possibly. Also, would always enabling this break things? Maybe we don't need that switch at all?

Reply:
If this is not intended, you may end up with conflicting responders on the network.. IIRC (I've pasted the quote somewhere) RFC allows several responders announcing the same information. I wonder if we should distinguish between the cases of the same information and different information.

Reply:
I agree that this is needed.

Is easy to make it backwards compatible as you say.

So we should either do that or change it to private.

From my perspective, this is a private method, I can’t see any reason to call it direct as it essentially registers the name on the network, so it only makes sense as part of calling register_service()

Reply:
I've updated so it is backwards compatible and we can have the debate about private methods in another PR!

Reply:
Yeah, sounds good